specs/ClearBrowsingData.md

```diff
@@ -0,0 +1,233 @@
+
+
+# Description
+The clear browsing data API is an asynchronous API. 
+The browsing data kinds that are supported are listed below.  These data kinds follow a hierarchical structure in which nested bullet points are included in their parent bullet point's data kind. 
+
+Ex: Dom storage is encompassed in site data which is encompassed in the profile data. Each of the following bullets correspond to a COREWEBVIEW2_BROWSING_DATA_KIND unless otherwise specified. 
+
+* Profile 
+    * Site Data
+        * Dom Storage 
+            * App Cache
+            * File Systems
+            * Indexddb
+            * Local Storage
+            * Web SQL
+            * Cache Storage
+            * Embedder Dom Storage (this API does not support this specific data kind but this is included in the dom storage data kind)
+            * Background Fetch (this API does not support this specific data kind but this is included in the dom storage data kind)
+        * Cookies
+        * Media Licenses
+        * Plugin Data
+        * Site Usage
+        * Durable Permissions
+        * External Protocols
+        * Isolated Origins
+        * Trust Tokens
+        * Conversions Data
+    * Http Cache 
+    * Download History
+    * General Autofill
+    * Password Autofill
+    * Bookmarks
+    * Settings
+    * Content Settings
+    * Local Custom Dictionary
+
+
+# Examples
+
+## Win32 C++
+```cpp
+/// Function to clear the password form data 
+bool EnvironmentComponent::ClearPasswordAutofillData() 
+{
+    wil::com_ptr<ICoreWebView2Environment5> environment;
+    webView->get_Environment(&environment);
+    CHECK_FAILURE(environment->ClearBrowsingData(COREWEBVIEW2_BROWSING_DATA_KIND_PASSWORD_AUTOFILL,
+        Callback<ICoreWebView2ClearBrowsingDataHandler>(
+            [this](HRESULT error, bool success) -> HRESULT {
+                if (success){
+                    return true;
+                } else {
+                    return false;
+                }
+                return S_OK;
+            })
+            .Get()));
+}
+```
+
+### .NET, WinRT
+```c#
+webView2Control.NavigationStarting += ClearPasswordAutofillData;
+
+private void ClearPasswordAutofillData() 
+{
+    var environment = webView2Control.CoreWebView2.Environment;
+    try
+    {
+        await environment.ClearBrowsingDataAsync(COREWEBVIEW2_BROWSING_DATA_KIND_PASSWORD_AUTOFILL); 
+    }
+    catch (System.Runtime.InteropServices.COMException exception)
+    {
+        // An exception occured
+    }
+}
+
+```
+
+# API Notes
+See [API Details](#api-details) section below for API reference.
+
+
+# API Details
+
+## Win32 C++
+
+```IDL
+interface ICoreWebView2Environment5
+interface ICoreWebView2ClearBrowsingDataCompletedHandler;
+
+/// Specifies the datatype for the 
+/// `ICoreWebView2Environment5::ClearBrowsingData` method.
+[v1_enum]
+typedef enum COREWEBVIEW2_BROWSING_DATA_KIND {
+  /// Specifies app cache data.
```

Comment:
Calling this 'app cache data' is sort of vague for folks unaware that there is a DOM feature named this. How about 
```suggestion
  /// Specifies data stored by the AppCache DOM feature.
```